### PR TITLE
fix: removed edit option of cell in locked view

### LINKED ISSUE #6266: 🐛 Bug: Locked view- double click on column header opens add/edit modal
### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

Screenshot 2023-08-29 at 8 33 48 PM

### Desired Behavior

Locked view should not allow column schema edit

### Project Details

Node: **v16.14.0**
Arch: **arm64**
Platform: **darwin**
Docker: **false**
RootDB: **pg**
PackageVersion: **0.111.0**

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by sreehari2003**
## Change Summary

Provide summary of changes with issue number if any.
- closes #6266 

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/header/Cell.vue [hunk 1]
```diff
@@ -18,6 +18,10 @@ const isDropDownOpen = ref(false)
 
 const isKanban = inject(IsKanbanInj, ref(false))
 
+const activeInj = inject(ActiveViewInj, ref())
+
+const isLockedView = computed(() => activeInj.value?.lock_type === 'locked')
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/header/Cell.vue:23]: you can just use `IsLockedInj` (from `#imports` also) directly.

```ts
const isLocked = inject(IsLockedInj, ref(false))
```

File: packages/nc-gui/components/smartsheet/header/Cell.vue [hunk 2]
```diff
@@ -39,6 +43,7 @@ const closeAddColumnDropdown = () => {
 }
 
 const openHeaderMenu = () => {
+  if (isLockedView.value) return
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/header/Cell.vue:46]: then simply put it together

```ts
if (!isLocked.value && !isForm.value && isUIAllowed('edit-column')) {
```

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: packages/nc-gui/components/smartsheet/header/Cell.vue [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <script setup lang="ts">
 import type { ColumnReqType, ColumnType } from 'nocodb-sdk'
-import { ActiveViewInj, ColumnInj, IsFormInj, IsKanbanInj, inject, provide, ref, toRef, useUIPermission } from '#imports'
+import { IsFormInj, IsKanbanInj, IsLockedInj, inject, provide, ref, toRef, useUIPermission } from '#imports'
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/header/Cell.vue:3]: seems you removed `ColumnInj` by mistake.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: packages/nc-gui/components/smartsheet/header/Cell.vue (hunk 3 above)
Comment (sreehari2003) [reply packages/nc-gui/components/smartsheet/header/Cell.vue]: fixed it , but removing that import didn't cause any error any idea why ?

**Comment by sreehari2003** [comment]
> @sreehari2003 Could see webhook option; has come back. needs a fix
> 
> Screenshot 2023-08-30 at 4 42 45 PM

ok

**Comment by dstala** [comment]
on hover over column header in locked view - cursor changes to clickable icon. it shouldn’t change.

**Comment by dstala** [comment]
& ignore https://github.com/nocodb/nocodb/pull/6268#issuecomment-1698969184
this is fixed in EE

**Review by dstala — APPROVED** [review]

**Review by wingkwong — APPROVED** [review]

**Pull request merged**